Error: reference to free variable `haskell-import-mapping'

Hey!

After updating to today git, I get this error when running `make`:

```
Source file `/home/bernat/.emacs.d/vendor/haskell-mode/haskell-session.el' newer than byte-compiled file
Source file `/home/bernat/.emacs.d/vendor/haskell-mode/haskell-utils.el' newer than byte-compiled file
Source file `/home/bernat/.emacs.d/vendor/haskell-mode/haskell-string.el' newer than byte-compiled file
Source file `/home/bernat/.emacs.d/vendor/haskell-mode/haskell-customize.el' newer than byte-compiled file

In toplevel form:
haskell-modules.el:43:32:Error: reference to free variable `haskell-import-mapping'
Makefile:84: recipe for target 'haskell-modules.elc' failed
make: *** [haskell-modules.elc] Error 1
```

Solved by `make clean`. I suppose some implicit rule `.elc.el` may help?

--- a/Makefile
+++ b/Makefile
@@ -28,7 +28,8 @@ EMACS := $(shell echo "$${EMACS:-emacs}")
 EFLAGS = --eval "(add-to-list 'load-path (expand-file-name \"tests/compat\") 'append)" \
 	 --eval "(when (< emacs-major-version 24) \
 		    (setq byte-compile-warnings '(not cl-functions)))" \
-	 --eval '(setq byte-compile-error-on-warn t)'
+	 --eval '(setq byte-compile-error-on-warn t)' \
+	 --eval '(when (not (version< emacs-version "24.4")) (setq load-prefer-newer t))'
 
 BATCH = $(EMACS) $(EFLAGS) --batch -Q -L .
 

